modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/stripper_outfit.dm

```diff
@@ -9,20 +9,72 @@
 	worn_icon_taur_hoof = 'modular_skyrat/modules/modular_items/lewd_items/icons/mob/lewd_clothing/lewd_uniform/lewd_uniform-hoof.dmi'
 	mutant_variants = STYLE_DIGITIGRADE|STYLE_TAUR_ALL
 	can_adjust = FALSE
-	icon_state = "stripper_cyan"
-	inhand_icon_state = "stripper_cyan"
-	unique_reskin = list("Cyan" = "stripper_cyan",
-						"Yellow" = "stripper_yellow",
-						"Green" = "stripper_green",
-						"Red" = "stripper_red",
-						"Latex" = "stripper_latex",
-						"Orange" = "stripper_orange",
-						"White" = "stripper_white",
-						"Purple" = "stripper_purple",
-						"Black" = "stripper_black",
-						"Black-teal" = "stripper_tealblack")
-
-/obj/item/clothing/under/stripper_outfit/AltClick(mob/user)
+	icon_state = "stripper"
+	inhand_icon_state = "stripper"
+	var/current_color = "cyan"
+	var/color_changed = FALSE
+	var/static/list/stripper_designs
+
+//create radial menu
+/obj/item/clothing/under/stripper_outfit/proc/populate_stripper_designs()
+	stripper_designs = list(
+		"cyan" = image (icon = src.icon, icon_state = "stripper_cyan"),
+		"yellow" = image(icon = src.icon, icon_state = "stripper_yellow"),
+		"green" = image(icon = src.icon, icon_state = "stripper_green"),
+		"red" = image(icon = src.icon, icon_state = "stripper_red"),
+		"latex" = image(icon = src.icon, icon_state = "stripper_latex"),
+		"orange" = image(icon = src.icon, icon_state = "stripper_orange"),
+		"white" = image(icon = src.icon, icon_state = "stripper_white"),
+		"purple" = image(icon = src.icon, icon_state = "stripper_purple"),
+		"black" = image(icon = src.icon, icon_state = "stripper_black"),
+		"tealblack" = image(icon = src.icon, icon_state = "stripper_tealblack"))
```

Comment:
really this should all be a component or something, it's repeated so many times almost identically